src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -141,11 +141,11 @@ class TaskReplaceActor(
         logger.warn(s"Deployment $deploymentId: New $id is terminal ($condition) on agent $agentId during app $pathId restart: " +
           s"$condition reservation: ${instance.reservation}. Waiting for the task to restart...")
         instanceTerminated(id)
-        instancesStarted -= 1
       } // 2) Did someone tamper with new instance's goal? Don't do that - there should be only one "orchestrator" per service per time!
       else if (considerTerminal(condition) && goal.isTerminal()) {
         logger.error(s"Deployment $deploymentId: New $id is terminal ($condition) on agent $agentId during app $pathId restart " +
           s"(reservation: ${instance.reservation}) and the goal ($goal) is *NOT* Running! This means that someone is interfering with current deployment!")
+        instancesStarted -= 1
```

Comment:
We would still have to check that we are not subtracting twice for the same instance id. This is a weird state anyways.